contract/test/tribbles-airdrop/e2e.test.js

```diff
@@ -0,0 +1,217 @@
+
+const bundleRoots = {
+  tribblesAirdrop: nodeRequire.resolve('../../src/airdrop.contract.js'),
+};
+
+const scriptRoots = {
+  tribblesAirdrop: nodeRequire.resolve('../../src/airdrop.proposal.js'),
+};
+
+/** @param {import('ava').ExecutionContext} t */
+const makeTestContext = async t => {
+  const bc = await makeBundleCacheContext(t);
+
+  const { E2E } = ambientEnv;
+  const { execFileSync, execFile } = ambientChildProcess;
+  const { writeFile } = ambientFsp;
+
+  /** @type {import('../../tools/agd-lib.js').ExecSync} */
+  const dockerExec = (file, args, opts = { encoding: 'utf-8' }) => {
+    const workdir = '/workspace/contract';
+    const execArgs = ['compose', 'exec', '--workdir', workdir, 'agd'];
+    opts.verbose &&
+      console.log('docker compose exec', JSON.stringify([file, ...args]));
+    return execFileSync('docker', [...execArgs, file, ...args], opts);
+  };
+
+  console.time('makeTestTools');
+  console.timeLog('makeTestTools', 'start');
+  // installBundles,
+  // runCoreEval,
+  // provisionSmartWallet,
+  // runPackageScript???
+  const tools = await (E2E
+    ? makeE2ETools(t, bc.bundleCache, {
+        execFileSync: dockerExec,
+        execFile,
+        fetch,
+        setTimeout,
+        writeFile,
+      })
+    : makeMockTools(t, bc.bundleCache));
+  console.timeEnd('makeTestTools');
+
+  return { ...tools, ...bc };
+};
+
+test.before(async t => (t.context = await makeTestContext(t)));
+
+test.serial('well-known brand (ATOM) is available', async t => {
+  const { makeQueryTool } = t.context;
+  const hub0 = makeAgoricNames(makeQueryTool());
+  const agoricNames = makeNameProxy(hub0);
+  await null;
+  const brand = {
+    ATOM: await agoricNames.brand.ATOM,
+  };
+  t.log(brand);
+  t.is(passStyleOf(brand.ATOM), 'remotable');
+});
+
+test.serial('install bundle: airdrop / tribblesAirdrop', async t => {
+  const { installBundles } = t.context;
+  console.time('installBundles');
+  console.timeLog('installBundles', Object.keys(bundleRoots).length, 'todo');
+  const bundles = await installBundles(bundleRoots, (...args) =>
+    console.timeLog('installBundles', ...args),
+  );
+
+  console.timeEnd('installBundles');
+
+  const id = getBundleId(bundles.tribblesAirdrop);
+  const shortId = id.slice(0, 8);
+  t.log('bundleId', shortId);
+  t.is(id.length, 3 + 128, 'bundleID length');
+  t.regex(id, /^b1-.../);
+  console.groupEnd();
+  Object.assign(t.context.shared, { bundles });
+});
+const containsSubstring = (substring, string) =>
+  new RegExp(substring, 'i').test(string);
+
+test.serial('deploy contract with core eval: airdrop / airdrop', async t => {
+  const { runCoreEval } = t.context;
+  const { bundles } = t.context.shared;
+  const bundleID = getBundleId(bundles.tribblesAirdrop);
+
+  t.deepEqual(
+    containsSubstring(bundles.tribblesAirdrop.endoZipBase64Sha512, bundleID),
+    true,
+  );
+  // this runCoreEval does not work
+  const name = 'airdrop';
+  const result = await runCoreEval({
```

Comment:
@dckc this CoreEval test seems to still fail

Reply:
symptoms? steps to reproduce?

Reply:
the errors stem from noble-hashes. I can't pinpoint exactly why the error is happening, but I've just recorded [this video](https://very-decentralized.myfilebase.com/ipfs/Qmd37r9YSapzz2YQfkVDxc1MvyMRZ7SdqVDTXTk9uDeTkU) where I speak on the issue in a fairly detailed manner.

The steps to reproduce are:

### preliminary steps
```
git fetch <remote-name> tg-deployment-test
cd contract
# check that all deps are installed and everything is working as expected
yarn run test -t
# see that output from test matches test output included in block below
```

#### test output
`yarn run test -t` should end with the following output:
```
1..25
# tests 24
# pass 18
# skip 5
# fail 2
```
If it doesn't, then I suggest running a `yarn install` from the root directory and then retrying. If that doesn't work then let me know and I'll try to figure out a way to get a copy of the proper code over to your machine.

## steps to reproduce error 

```sh
$ cd contract
$ docker-compose up -d
$ yarn run docker:logs
# open up new terminal window
$ make clean start-contract-airdrop
```

The proposal will pass and then the CoreEval will fail, producing the error below:


```sh
agd-1  | 2024-09-02T20:50:18.308Z SwingSet: vat: v1: CORE_EVAL failed: (TypeError#8)
agd-1  | 2024-09-02T20:50:18.309Z SwingSet: vat: v1: TypeError#8: toBytes: no function
agd-1  | 2024-09-02T20:50:18.309Z SwingSet: vat: v1: TypeError: toBytes: no function
agd-1  |  at toBytes (#11989:331)
agd-1  |  at hashC (#11989:343)
agd-1  |  at reduceRight ()
agd-1  |  at map ()
agd-1  |  at generateMerkleRoot (#11989:791)
agd-1  |  at (#11989:981)
agd-1  |  at (#11988:11)
agd-1  |  at apply ()
agd-1  |  at safeEvaluate (/bundled-source/.../node_modules/ses/src/make-safe-evaluator.js:78)
agd-1  |  at (.../vats/src/core/chain-behaviors.js:88)
agd-1  |  at ()
agd-1  |
agd-1  | 2024-09-02T20:50:18.310Z SwingSet: xsnap: v1: TypeError#8 ERROR_NOTE: Sent as error:liveSlots:v1#70006
agd-1  | 2024-09-02T20:50:18.311Z SwingSet: ls: v1: Logging sent error stack (TypeError#8)
agd-1  | 2024-09-02T20:50:18.319Z SwingSet: xsnap: v10: UnhandledPromiseRejectionWarning: (RemoteTypeError(error:liveSlots:v1#70006)#10)
agd-1  | 2024-09-02T20:50:18.319Z SwingSet: xsnap: v10: RemoteTypeError(error:liveSlots:v1#70006)#10: toBytes: no function
agd-1  | 2024-09-02T20:50:18.321Z SwingSet: xsnap: v10: TypeError: toBytes: no function
agd-1  |  at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
agd-1  |  at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:281)
agd-1  |  at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:434)
agd-1  |  at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:356)
agd-1  |  at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1225)
agd-1  |  at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1249)
agd-1  |  at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1514)
agd-1  |  at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:60)
agd-1  |  at ()
```

### brandAuxPublisher failure
this error is shown in the video. it appears before the CoreEval errors, and appears to be separate from the CoreEval failure, but I'm adding the stack trace as its still relevant.

```
agd-1  | 2024-09-02T20:50:01.000Z block-manager: block 6617 begin
agd-1  | 2024-09-02T20:50:01.067Z SwingSet: vat: v1: brandAuxPublisher: new Promise
agd-1  | 2024-09-02T20:50:01.068Z SwingSet: vat: v1: boardAuxAdmin: new Promise
agd-1  | 2024-09-02T20:50:01.072Z SwingSet: vat: v1: cannot save non-durable: brandAuxPublisher Object [Alleged: BrandAuxPublisher] { publishBrandInfo: [Function: publishBrandInfo] }
agd-1  | 2024-09-02T20:50:01.073Z SwingSet: vat: v1: brandAuxPublisher settled; remaining: [ 'anchorKits', 'bankMints', 'boardAuxAdmin', 'contractGovernor', 'reserveGovernorCreatorFacet', 'scaledPriceAuthorityKits', 'vBankKits', 'vaultFactoryGovernorCreator' ]
agd-1  | 2024-09-02T20:50:01.074Z SwingSet: vat: v1: cannot save non-durable: boardAuxAdmin Object [Alleged: BoardAuxAdmin] { publishBrandInfo: [Function: publishBrandInfo], init: [Function: init], update: [Function: update] }
agd-1  | 2024-09-02T20:50:01.075Z SwingSet: vat: v1: boardAuxAdmin settled; remaining: [ 'anchorKits', 'bankMints', 'contractGovernor', 'reserveGovernorCreatorFacet', 'scaledPriceAuthorityKits', 'vBankKits', 'vaultFactoryGovernorCreator' ]
```

Reply:
> ...
> The proposal will pass and then the CoreEval will fail, producing the error below:
> 
> ```
> ...
> agd-1  |  at generateMerkleRoot (#11989:791)
> ...
> agd-1  |  at safeEvaluate (/bundled-source/.../node_modules/ses/src/make-safe-evaluator.js:78)
> agd-1  |  at (.../vats/src/core/chain-behaviors.js:88)
> ```

Why is there a call to `generateMerkleRoot` in the CoreEval script? `generateMerkleRoot` takes the large collection of  public keys representing eligible participants as an argument, right? The core-eval script goes on chain. It cannot contain the large collection of keys, right? The merkle root should be generated in advance and provided to the core-eval script as a configuration argument, right?

See also:

 - #72

Reply:
yes, updating now..

Reply:
@dckc this has been addressed in https://github.com/hindley-milner-systems/dapp-ertp-airdrop/pull/99/commits/fcb20318a75bb1be4c5c52e2a20fd57c18c34b88.

the test code has been updated accordingly.